what-is-pair.org

```diff
@@ -0,0 +1,251 @@
+#+title: What is Pair?
+#+date: <2022-08-08 Mon>
+
+A description of our Pair service. This is not a description of "what should
+or could be", but how Pair exists now.
+
+* Key Terms
+- Pair :: This refers to the code and service held at
+  https://github.com/sharingio/pair, at the time of this writing and to the code
+  and customizations held at https://github.com/sharingio/.sharing.io. That
+  means, specifically, the code (and resulting programs) from the codebase at
+  commit [[https://github.com/sharingio/pair/commit/a473a19c1c69cafc21d7660f7d101ebea134402f][a473a19c1c69cafc21d7660f7d101ebea134402f]] (for pair) and commit
+  [[https://github.com/sharingio/.sharing.io/commit/22eb981967244d7031548c1aa66e66f01b9a681b][22eb981967244d7031548c1aa66e66f01b9a681b]] for .sharing.io
+- Environment ::
+  The contained environment designed for the work that
+  happens when using Pair. It is a pod within the pair instance that we
+  configure for a highly opiniated form of development and pairing.
+  In this case, I am talking of code held at github.com/sharingio/environment,
+  at commit [[https://github.com/sharingio/environment/commit/f8dad1fe163151a4cae0bf4b962fb55a655416c0][f8dad1fe163151a4cae0bf4b962fb55a655416c0]]
+
+* What is Pair?
+Pair is a mechanism for creating kubernetes clusters on Equinix bare metal
+servers through a simple interface. These clusters will come with services and
+settings "pre-installed" to improve developer ergonomics, especially when pair
+programming with a remote pair.
+
+The services that are pre-installed on the cluster are a combination of ones:
+- chosen by the Pair team, to be  installed on any new cluster
+- chosen by the developer, specific to that cluster
+- chosen by projects the developer deploys, meant to be reproducible across Pair
+  clusters.
+
+** Services chosen by the Pair Team
+All new clusters on Pair come with:
+- Domain name management and TLS Support ::
+  This means any exposed cluster service can be given a recognizable, sharable
+  url and each follow a similar pattern. If I create a pair instance named 'cool' ,
+  and a namespace called 'fun', then all services i expose from that namespace will
+  have a url of ~{SERVICENAME}.fun.cool.pair.sharing.io~
```

Comment:
I think this is regarding the auto exposing of processes listening on a port on all interfaces.
If so, services listening will come up on `{SERVICENAME}.cool.pair.sharing.io`, service name being overridable via the `${SHARINGIO_PAIR_EXPOSER_HOSTNAME}` variable (see https://github.com/sharingio/environment/blob/9fac334/pkg/types/types.go#L51)

Reply:
resolved with [5c8345a](https://github.com/sharingio/org/pull/1/commits/5c8345aa1d3022b07c721db69384534138666adc)